src/lib/utils/hooks/SynapseAPI/forum/useForum.ts

```diff
@@ -224,3 +226,61 @@ export function useGetRepliesInfinite(
     },
   )
 }
+
+export function usePostReply(
+  options?: UseMutationOptions<
+    DiscussionReplyBundle,
+    SynapseClientError,
+    CreateDiscussionReply
+  >,
+) {
+  const queryClient = useQueryClient()
+  const { accessToken } = useSynapseContext()
+
+  return useMutation<
+    DiscussionReplyBundle,
+    SynapseClientError,
+    CreateDiscussionReply
+  >(
+    (request: CreateDiscussionReply) =>
+      SynapseClient.postReply(request, accessToken),
+    {
+      ...options,
+      onSuccess: async (newReply, variables, ctx) => {
+        await queryClient.invalidateQueries(['thread'])
```

Comment:
...then you can invalidate _just_ the thread that was updated here
```suggestion
        await queryClient.invalidateQueries(['thread', request.threadId])
```